### PR TITLE
feat: implement `OnyxVisuallyHidden` component

### PR DESCRIPTION
**Pull request by JoCa96**
Relates to #1083 

- Implemented `OnyxVisuallyHidden` component to be able to add assistive-technology-only text
- Used new component to add `opens in new tab` information to `OnyxLink` component
- Added label to `OnyxNavAppArea` button for improved accessibility

### CODE REVIEW

File: packages/sit-onyx/src/components/OnyxVisuallyHidden/OnyxVisuallyHidden.vue
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxVisuallyHidden/OnyxVisuallyHidden.vue:1]: Would prefer to implement it as class instead of component so its easier to apply it with conditions like:

`<span :class="{'onyx-visually-hidden': props.test === true}">Test</span>`

File: packages/sit-onyx/src/i18n/locales/de-DE.json [hunk 1]
```diff
@@ -27,9 +27,13 @@
...
     "clearSearch": "Suchtext löschen"
   },
+  "link": {
+    "opensExternally": "Wird in einem neuen Tab geöffnet."
+  },
   "navItemOptionsLabel": "Unterseiten von {label}",
   "navigation": {
-    "appLogo": "App logo",
+    "appLogo": "App logo von {appName}",
+    "goToHome": "Gehe zur Startseite",
```
Comment (larsrickert) [inline packages/sit-onyx/src/i18n/locales/de-DE.json:36]: Technically the app area click is up to the project so it does not have to go to the home page

File: packages/sit-onyx/src/components/OnyxLink/OnyxLink.vue [hunk 2]
```diff
@@ -24,21 +26,20 @@ defineSlots<{
...
 
 <template>
-  <a
-    class="onyx-link"
-    :href="props.href"
-    :target="props.target"
-    :rel="props.target === '_blank' ? 'noreferrer' : undefined"
-    @click="emit('click')"
-  >
+  <a class="onyx-link" :href="props.href" :target="props.target" @click="emit('click')">
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxLink/OnyxLink.vue:38]: Why did you remove the `rel`? The browser only infers `noopener`but not `noreferrer`

File: packages/sit-onyx/src/components/OnyxNavAppArea/types.ts [hunk 3]
```diff
@@ -7,4 +7,8 @@ export type OnyxNavAppAreaProps = {
    * Application name to show on the left.
    */
   appName?: string;
+  /**
+   * Describes the action that is taken on click.
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxNavAppArea/types.ts:11]:
```suggestion
   * Describes the action that is taken on click.
   * By default, a translated "Go to home" will be used depending on the current locale.
```

File: packages/sit-onyx/src/components/OnyxNavBar/types.ts [hunk 4]
```diff
@@ -5,4 +5,9 @@ export type OnyxNavBarProps = OnyxNavAppAreaProps & {
    * Whether to show a back button.
    */
   withBackButton?: boolean;
+  /**
+   * Custom label for the app click area.
+   * @default "Go to Home"
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxNavBar/types.ts:10]:
```suggestion
   * @default Translated "Go to home" depending on the current locale.
```

File: packages/sit-onyx/src/components/OnyxVisuallyHidden/OnyxVisuallyHidden.ct.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import { expect, test } from "../../playwright/a11y";
+import OnyxVisuallyHidden from "./OnyxVisuallyHidden.vue";
+
+test("should not be visible", async ({ mount }) => {
+  const component = await mount(
+    <div>
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxVisuallyHidden/OnyxVisuallyHidden.ct.tsx:6]:
```suggestion
    <div style={{ width: "max-content" }}>
```

So the screenshots are smaller and easier to read

File: packages/sit-onyx/src/components/OnyxVisuallyHidden/types.ts [hunk 6]
```diff
@@ -0,0 +1,3 @@
+export type OnyxVisuallyHiddenProps = {
+  is?: string;
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxVisuallyHidden/types.ts:2]: Would prefer to make it required so its more explicit

File: packages/sit-onyx/src/i18n/locales/en-US.json [hunk 7]
```diff
@@ -27,9 +27,13 @@
...
     "clearSearch": "Clear search filter"
   },
+  "link": {
+    "opensExternally": "Opens in a new tab."
+  },
   "navItemOptionsLabel": "Subpages of {label}",
   "navigation": {
-    "appLogo": "App logo",
+    "appLogo": "App logo of {appName}",
+    "goToHome": "Go to Home",
```
Comment (larsrickert) [inline packages/sit-onyx/src/i18n/locales/en-US.json:36]:
```suggestion
    "goToHome": "Go to home page",
```

**Review by larsrickert — CHANGES_REQUESTED** [review]

File: packages/sit-onyx/src/i18n/locales/de-DE.json (hunk 1 above)
Comment (JoCa96) [reply packages/sit-onyx/src/i18n/locales/de-DE.json:36]: Yes, that's why the label is configurable. But we could also make the label required, if you prefer

File: packages/sit-onyx/src/components/OnyxVisuallyHidden/types.ts (hunk 6 above)
Comment (JoCa96) [reply packages/sit-onyx/src/components/OnyxVisuallyHidden/types.ts:2]: Usually there should not be a need to change the element.

File: packages/sit-onyx/src/components/OnyxVisuallyHidden/OnyxVisuallyHidden.vue
Comment (JoCa96) [reply packages/sit-onyx/src/components/OnyxVisuallyHidden/OnyxVisuallyHidden.vue:1]: This component is not something that should be toggled.

File: packages/sit-onyx/src/components/OnyxNavBar/types.ts [hunk 8]
```diff
@@ -5,4 +5,9 @@ export type OnyxNavBarProps = OnyxNavAppAreaProps & {
    * Whether to show a back button.
    */
   withBackButton?: boolean;
+  /**
+   * Custom label for the app click area.
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxNavBar/types.ts:9]:
```suggestion
   * Custom (aria) label for the app click area.
```

File: packages/sit-onyx/src/components/OnyxVisuallyHidden/OnyxVisuallyHidden.vue [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+.onyx-visually-hidden {
+  @include layers.component() {
+    border: 0;
+    clip: rect(0, 0, 0, 0);
+    height: 0;
+    margin: 0;
+    overflow: hidden;
+    padding: 0;
+    position: absolute;
+    width: 1px;
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxVisuallyHidden/OnyxVisuallyHidden.vue:23]:
```suggestion
    width: var(--onyx-1px-in-rem);
```

**Review by larsrickert — CHANGES_REQUESTED** [review]

File: packages/sit-onyx/src/components/OnyxVisuallyHidden/OnyxVisuallyHidden.vue (hunk 9 above)
Comment (JoCa96) [reply packages/sit-onyx/src/components/OnyxVisuallyHidden/OnyxVisuallyHidden.vue:23]: The component should have the smallest possible width, so that it is still recognized as a "visible" element, but it shouldn't scale with the root element font-size.

**Review by larsrickert — APPROVED** [review]
